dask_expr/_merge.py

```diff
@@ -1,11 +1,16 @@
 import functools
 
 from dask.dataframe.dispatch import make_meta, meta_nonempty
-from dask.utils import M, apply
+from dask.utils import M, apply, get_default_shuffle_method
+from distributed.shuffle._core import ShuffleId, barrier_key
+from distributed.shuffle._merge import merge_transfer, merge_unpack
+from distributed.shuffle._shuffle import shuffle_barrier
 
-from dask_expr._expr import Blockwise, Expr, Index, Projection
+from dask_expr._expr import Blockwise, Expr, Index, PartitionsFiltered, Projection
 from dask_expr._repartition import Repartition
-from dask_expr._shuffle import Shuffle, _contains_index_name
+from dask_expr._shuffle import AssignPartitioningIndex, Shuffle, _contains_index_name
+
+_HASH_COLUMN_NAME = "__hash_partition"
```

Comment:
I take it this is a "magic" string used by `distributed`'s p2p merge?

Reply:
Yep